File: bookshop/src/main/java/com/justlamvt05/bookshop/service/AuthService.java
package com.justlamvt05.bookshop.service;

import com.justlamvt05.bookshop.domain.dto.UserDto;
import com.justlamvt05.bookshop.payload.request.RegisterRequest;
import com.justlamvt05.bookshop.payload.response.ApiResponse;


public interface AuthService {
   UserDto login(String username, String password);
   ApiResponse<?> register(RegisterRequest request);
}

File: src/service/authService.js
import axios from "axios";

export const login = async (username, password) => {
  const response = await axios.post(
    "http://localhost:8080/api/auth/login",
    { username, password }
  );
  return response.data; // ApiResponse
};
export const register = async (payload) => {
  const response = await axios.post(
    "http://localhost:8080/api/auth/register",
    payload
  );
  return response.data; // ApiResponse
};


export const logout = () => {
  localStorage.removeItem("token");
};
